src/MessageQueue/MessageQueue.php

```diff
@@ -151,7 +151,9 @@ private function doFlush(bool $async)
             }
 
             if ($async) {
-                $this->sqsClient->sendMessageBatchAsync($parameters);
+                $promise = $this->sqsClient->sendMessageBatchAsync($parameters);
+
+                $promise->wait();
```

Comment:
Not sure to understand here. Isn't the goal of async especially not waiting and be in "fire and forget" mode?

Reply:
Sure but a promise should have a `then` to be launched and do something with the result, here as we do not need to get the result we have to ask the sdk to really send the commands.

Reply:
The commands are sent in parallel in batch of 25 AFAIK.